### DISCUSSION TITLE
Db2 Install .- Invalid HostName problems

### DISCUSSION THREAD

**Discussion by agorosti** [HCL-TECH-SOFTWARE/connections-automation#88] [General]
Hi,

I was using default values for installation,  and I had the following error when doing the quickstart install command.
It seems that the standar/example value used for the quickstart, was not valid.

fatal: [connections.internal.example.com]: FAILED! => {"changed": true, "cmd": ["/opt/IBM/InstallBinaries/universal/db2setup", "-r", "/opt/IBM/InstallBinaries/universal/db2server.rsp"], "delta": "0:00:07.753481", "end": "2021-03-21 09:08:04.055421", "msg": "non-zero return code", "rc": 87, "start": "2021-03-21 09:07:56.301940", "stderr": "", "stderr_lines": [], "stdout": "DBI1191I  db2setup is installing and configuring DB2 according to the\n      response file provided. Please wait.\n\n\nThe host name \"connections.internal.example.com\" is invalid. Specify a valid\nhost name.\n\nThe response file specified \"/opt/IBM/InstallBinaries/universal/db2server.rsp\"\nis not valid.\n\nFor more information see the DB2 installation log at \"/tmp/db2setup.32563.log\".", "stdout_lines": ["DBI1191I  db2setup is installing and configuring DB2 according to the", "      response file provided. Please wait.", "", "", "The host name \"connections.internal.example.com\" is invalid. Specify a valid", "host name.", "", "The response file specified \"/opt/IBM/InstallBinaries/universal/db2server.rsp\"", "is not valid.", "", "For more information see the DB2 installation log at \"/tmp/db2setup.32563.log\"."]}

Intially I though the error was due the hostname used  (having subdomains),  so I changed the name fora different one,  added that one also on /etc/hosts,   changed the playbook  and  I did a hostnamectl set-hostname   to change it on the machine,  but when I run the ansible script again,  I had the same error.

Any hint ?

Thks!

**Comment by agorosti** [comment]
Solved.  As an additional command,   I went to the connections.internal.example.com  host ,   logged as root,   and I did manualy a ./db2setup -r db2server.rsp  from the path than ansible created at /opt/IBM/InstallBinaries/universal  ,  

 I got the same error,  host name is invalid  I ensured the hostname on my machine had not FDQN.
I did  a  "sudo hostnamectl set-hostname connections",    and now it works.

I am using Red Hat Enterprise Linux release 8.3 (Ootpa)

**Reply by dejo1307** [reply]
Hi Andres,

I must say I didn't test it with RHEL 8.3 - only because I was pointed out that Connections doesn't support RHEL 8.3 officially and, from another new, OpenLDAP is not there either (also not on CentOS 8.3) so at least that step would break in this case (of course if you don't have OpenLDAP somewhere else, or any other directory server).

I do remember, from one internal installation, with RHEL specifically, some issues with hostnamectl that you just described, but I guess that is something RH specific since we are testing in 99% of cases on CentOS.

I'll check next week with RHEL 8.3 as well, but I would say here, for CNX part at least, it makes sense to fall back to RHEL 7.9/CentOS 7.9 if possible, simply to have it working end to end (because OpenLDAP).

Best,
Dejan

P.S. Will push some fixes next week related to CentOS/RHEL 8.3 for CP stack. That should work wherever you can run Kubernetes, but there is at least one fix that I just need to push which would break Docker installation on 8.3 in this very moment which is not pushed yet.

**Reply by agorosti** [reply]
Dejan,  due the issues with LDAP  U  I re-did the install on Centos 7.9,  and I had the same issue with hostnames.

It was solved when I specifically did a hostnamectl set-hostname  "shortname"   ,   so it seems that with Centos 7.9 the problem with hostnames also happens.

By the way,  I was unable to install dockers  on a RHEL image on AWS  (since docker can be installed via repositories on RedHat, and a subscription is described) so I guess I will use also RHEL for CP,   not sure if your fix will deal aslo with Redhat images on amazon that have not enabled the subscription,    and I guess it will be easier for me to have Centos instead.
